mbed-hal/spi_api.h

```diff
@@ -32,10 +32,33 @@
 #define SPI_FILL_WORD         (0xFFFFFFFF)
```

Comment:
I want to comment on the `#define SPI_EVENT*` up there, but GitHub won't let me.

Anyways, I will refactor this into an enum, so that `spi_irq_handler_asynch` returns `spi_event_t`.
Is that ok? We are abusing an enum as a bit field here.